### ISSUE TITLE
Map Viewport doesn't always update according to ViewState

### ISSUE THREAD

**Issue by marioaks** [visgl/deck.gl#4513]
Hey there,

### Description
There seems to be a problem with the OrthographicView viewport not always reflecting the viewState prop when my map loads. 

Essentially, I am trying to dynamically control the viewState to constrain my view to a bounding box. A simplified version of my code is the following:

```
const Map = ({maxBounds}) => {
    const [viewState, setViewState] = useState(null)
    const [isMapLoaded, setIsMapLoaded] = useState(false)

   useEffect(() => {
    if (isMapLoaded) {
        const {minZoom, target} = calculateInitialViewState(maxBounds)
       setViewState(vs => ({...vs, zoom: minZoom, minZoom, target}))
    }
  }, [isMapLoaded, maxBounds] )

  const onViewStateChange = ({viewState: vs}) => {
    const {target, zoom} = calculateBoundedViewState(vs, maxBounds)
    setViewState({ ...vs, target, zoom })
  }

  return (
    <DeckGL
        onLoad={() => setIsMapLoaded(true)}
        viewState={viewState}
        onViewStateChange={onViewStateChange}
       views={new OrthographicView()}
       controller={true}
    >
        {layers}
    </DeckGL>
  )
}
```

The problem occurs about 20% of the times I reload the page. I'll console.log what the current viewState is, and about 20% of the time, the viewport does not reflect that. It seems like the times it doesn't work, the viewManager has a an `{}` as it's viewState, and therefore what i see is a viewport centered at `target: [0,0]` and `zoom: 0`. 

It looks to me like a lot of changes were made with viewState / initialViewState around version 8.1 that still don't seem to be super clearly documented. Did something change that might be causing this bug?

I noticed a similar issue (https://github.com/visgl/deck.gl/issues/4489#issue-599475498), but since it is closed I figured I would open a new one. 

Thanks!

### Environment
Framework Version: [e.g. deck.gl 8.2.0-alpha.1]
Browser Version: [e.g. Chrome 81.0]
OS: [e.g. Mac OS X 10.14]

**Comment by Pessimistress** [comment]
`viewManager.viewState` would only be `{}` if a falsy `viewState` is passed in. Your initial value for `viewState` is `null`. Is there a reason you have to calculate the initial value after DeckGL is loaded?

**Comment by marioaks** [comment]
It doesn’t seem like the problem is in initially passing in a falsy value as the `viewState.` I tried initializing `viewState` with a real viewport object instead of `null` and this bug persisted. Whether or not the initial `viewState` is falsy or not, the `viewManager.viewState` doesn’t always update correctly whenever my `viewState` variable changes.  

So, I ended up fixing this by preventing Deck from rendering until the initial `viewState` was set. I basically just added in a `{!!viewState && <DeckGL ... >}` line to my code and removed the `onLoad` logic. It doesn’t seem like a great solution, but so far it’s fixed the problem. This way, the viewManager doesn’t have to listen to any `viewState` changes during its first render. 

I guess what I’m trying to figure out now is, why does this have to be the case for it to work? My expectation would be that the viewManager would always listen to and update according to the current `viewState` variable. It is inconsistently failing at doing that around initialization.

**Comment by lukaszbarkowski** [comment]
@Pessimistress 

Having the same problem. When I fetch some data, I want to first reset the viewState to default values and when the request succeeds I want to set a new viewState. The issue occures randomly, more often on slower machines or with set CPU throttling.

Demo here: [https://codesandbox.io/s/eager-galileo-ivgwz](url)
It may take some clicks to reproduce.

Browser: Chrome 83.0.4103.116 / Firefox 77.0.1
Deck version: 8.2.0

**Comment by Pessimistress** [comment]
@lukaszbarkowski what am I looking for in your demo?

**Comment by lukaszbarkowski** [comment]
@Pessimistress Sometimes the map doesn't reflect the viewState, you can click fetch button, the viewState will reset, but when the request succeds, so we have a new viewState but the view isn't updated and it's stuck on the default one. The viewState is actually a second console.log, but deck.gl represents the previous one with the zoom 3:
![image](https://user-images.githubusercontent.com/40040662/86642626-56e85b00-bfdc-11ea-8dc6-81589507727c.png)

**Comment by Pessimistress** [comment]
@lukaszbarkowski I don't think your app reproduces the original issue. You use many layers of hooks that may be asynchronous and conflict with each other.

You need to move the `console.log` to right before `<DeckGL>` is returned, and see if the `viewState` object reflects deck's viewport. If `viewState` is not what you expect, then the issue is elsewhere.

**Comment by lukaszbarkowski** [comment]
@Pessimistress It looks like it is passed to `<DeckGL>` correctly, but for some reason the original transition to default state doesn't stop. I don't specify `transitionInterruption` so I guess it should stop the actual one and start the new one ?

[Updated demo with console.logs](https://codesandbox.io/s/eager-galileo-ivgwz)
![image](https://user-images.githubusercontent.com/40040662/86734004-f1bf5480-c031-11ea-9642-c0e6b84a31c6.png)

**Comment by Pessimistress** [comment]
This will be fixed in 8.5.0. You can try it right now with `8.5.0-alpha.1`.
